src/BenchmarkDotNet/Running/BenchmarkRunnerDirty.cs

```diff
@@ -16,18 +20,22 @@ namespace BenchmarkDotNet.Running
     // so it can ignore the version mismatch ;)
     public static class BenchmarkRunner
     {
+        private static readonly IUserInteraction userInteraction = new UserInteraction();
+        private static readonly List<Type> types = new List<Type>();
+        private static readonly List<Assembly> assemblies = new List<Assembly>();
+
         [PublicAPI]
-        public static Summary Run<T>(IConfig config = null)
+        public static Summary Run<T>(IConfig config = null, string[] args = null)
         {
             using (DirtyAssemblyResolveHelper.Create())
-                return RunWithExceptionHandling(() => RunWithDirtyAssemblyResolveHelper(typeof(T), config));
+                return RunWithExceptionHandling(() => RunWithDirtyAssemblyResolveHelper(typeof(T), config, args ?? Array.Empty<string>()));
         }
 
         [PublicAPI]
-        public static Summary Run(Type type, IConfig config = null)
+        public static Summary Run(Type type, IConfig config = null, string[] args = null)
```

Comment:
could you please extend also an overload that takes an `Assembly` as an argument?

`public static Summary[] Run(Assembly assembly, IConfig config = null)`

Reply:
you mean we need on more,
```  public static Summary[] Run(Assembly assembly, IConfig config = null,string[] args = null) ```
that takes assembly, config and args.

Reply:
> that takes assembly, config and arg

yes, exactly

Reply:
I am sorry for not making myself clear. Could you please extend the existing method with a new optional parameter?

https://github.com/dotnet/BenchmarkDotNet/blob/7c31942f6a78c73e14f9d1b7451d866540657139/src/BenchmarkDotNet/Running/BenchmarkRunnerDirty.cs#L52